### PR TITLE
Properly handle trivia when replacing a property with methods.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/roslyn-ide  review plz

**Comment by jmarolf** [comment]
:shipit:

**Comment by CyrusNajmabadi** [comment]
Also, properly handle rewrites when a property is referenced within its own write.  i.e. when you have: "this.Prop = this.Prop + 1".

**Comment by CyrusNajmabadi** [comment]
test vsi please

**Comment by CyrusNajmabadi** [comment]
retest windows_vsi_p2_open_prtest  please

File: src/Features/Core/Portable/ReplacePropertyWithMethods/AbstractReplacePropertyWithMethodsService.cs [hunk 1]
```diff
@@ -100,6 +102,81 @@ private struct ReferenceReplacer
...
+            //
+            // then when we hit "this.Prop" on the left of the equals, we'll want
+            // to see the results of the tree *after* we've replaced "this.Prop + 1"
+            // with "this.GetProp() + 1".  If we don't do this, and instead examine
+            // the "this.Prop" in the original tree, then we won't see that other
+            // rewrite.
+            //
+            // The SyntaxEditor API works by passing in these callbacks when we 
+            // replace a node N.  It will call us back with what N looks like after
+            // all teh rewrites that occurred underneath it.
```
Comment (Pilchie) [inline src/Features/Core/Portable/ReplacePropertyWithMethods/AbstractReplacePropertyWithMethodsService.cs]: teh

**Comment by Pilchie** [comment]
Nice. Thanks!  :+1:

**Pull request merged**